etcd/cmd/microshift-etcd/run.go

```diff
@@ -97,17 +112,66 @@ func (s *EtcdService) Run() error {
 	}
 	<-e.Server.ReadyNotify()
 
+	// If we were told to, go ahead and do a defragment now.
+	if s.doStartupDefrag {
+		if err := e.Server.Backend().Defrag(); err != nil {
+			err = fmt.Errorf("initial defragmentation failed: %v", err)
+			klog.Error(err)
+			return err
```

Comment:
In that case, should we put `e.Server.Stop()` and `<-e.Server.StopNotify()` in defer just after starting?

Reply:
Good catch